lib/Plack/Middleware/Deflater.pm

```diff
@@ -54,6 +54,9 @@ sub call {
             push @vary, 'User-Agent' if $self->vary_user_agent;
             $h->set('Vary' => join(",", @vary));
 
+            # Do not clobber already existing encoding
+            return if $h->exists('Content-Encoding') && $h->get('Content-Encoding') ne 'identity';
```

Comment:
You'll get an uninitialized warning if Content-Encoding doesn't exist, because of `&&` instead of `and`:

    ... if $h->exists('Content-Encoding') && $h->get('Content-Encoding') ne 'identity';
    ... if ($h->exists('Content-Encoding') && $h->get('Content-Encoding')) ne 'identity';
    ... if (!1 && undef) ne 'identity';
    ... if undef ne 'identity';